File: Java/infetionspread/Node.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


class Node(object):
    """ generated source for Node

    """
    susceptibility = float()
    targetSusceptibility = float()
    currentSusceptibility = float()
    infected = bool()
    vaccineEffectCompleted = bool()
    infectionTime = 0
    vaccineEffectiveness = 0.7
    infectionImmunity = 0.7
    susceptibilityRecoveryValue = 0.003
    recoveryTime = 5
    vaccineEffectPeriod = 14

    def getSusceptibility(self):
        return self.susceptibility

    def setSusceptibility(self, susceptibility):
        self.susceptibility = self.susceptibility

    def getTargetSusceptibility(self):
        return self.targetSusceptibility

    def setTargetSusceptibility(self, targetSusceptibility):
        self.targetSusceptibility = self.targetSusceptibility

    def getCurrentSusceptibility(self):
        return self.currentSusceptibility

    def setCurrentSusceptibility(self, currentSusceptibility):
        self.currentSusceptibility = self.currentSusceptibility

    def isInfected(self):
        return self.infected

    def setInfected(self, infected):
        self.infected = self.infected

    def isVaccineEffectCompleted(self):
        return self.vaccineEffectCompleted

    def setVaccineEffectCompleted(self, vaccineEffectCompleted):
        self.vaccineEffectCompleted = self.vaccineEffectCompleted

    def getInfectionTime(self):
        return self.infectionTime

    def setInfectionTime(self, infectionTime):
        self.infectionTime = self.infectionTime

    def getNeighbours(self):
        return neighbours

    def setNeighbours(self, neighbours):
        self.neighbours = neighbours

    def __init__(self):
        random = Random()
        self.susceptibility = random.nextInt(35) / 1000.00 + 0.016
        self.currentSusceptibility = self.targetSusceptibility = self.susceptibility
        self.infected = False
        self.vaccineEffectCompleted = True
        self.infectionTime = 0
        print "Value of susceptibility for node = " + self.susceptibility

    def vaccinate(self):
        self.vaccineEffectCompleted = False
        self.targetSusceptibility = self.susceptibility * 1.0 - self.vaccineEffectiveness

    def updateSusceptibility(self):
        if not self.vaccineEffectCompleted:
            if self.currentSusceptibility > self.targetSusceptibility:
                self.currentSusceptibility -= self.susceptibility * self.vaccineEffectiveness / self.vaccineEffectPeriod
                return True
            self.vaccineEffectCompleted = True
            self.targetSusceptibility = self.susceptibility
        if self.currentSusceptibility < self.targetSusceptibility:
            self.currentSusceptibility += self.susceptibilityRecoveryValue
            if self.currentSusceptibility > self.targetSusceptibility:
                self.currentSusceptibility = self.targetSusceptibility
            return True
        return False

    def infect(self, globalDecayValue):
        if self.isInfected():
            return False
        if random.nextInt(1001) / 1000.00 < self.getCurrentSusceptibility():
            self.setInfected(True)
            self.setInfectionTime(0)
            return True
        return False

    def isRecovered(self):
        if not self.isInfected():
            return False
        self.infectionTime += 1
        if self.infectionTime < self.recoveryTime:
            return False
        self.setInfected(False)
        self.resetInfection()
        return True

    def resetInfection(self):
        self.infectionTime = 0
        self.currentSusceptibility = self.currentSusceptibility * 1.0 - self.infectionImmunity
        self.targetSusceptibility = self.susceptibility
        self.vaccineEffectCompleted = True

File: Java/infetionspread/Node.java
package infetionspread;

import java.util.HashSet;
import java.util.Random;
import java.util.Set;

/**
 * Created with IntelliJ IDEA.
 * User: akash
 * Date: 10/27/13
 * Time: 10:20 PM
 * To change this template use File | Settings | File Templates.
 */
public class Node {
    private double susceptibility;
    private double targetSusceptibility;
    private double currentSusceptibility;
    private boolean infected;
    private boolean vaccineEffectCompleted;
    private int infectionTime;
    private Set<Integer> neighbours;

    private Random random;

    private final double vaccineEffectiveness = 0.7;
    private final double infectionImmunity = 0.02;
    private final double susceptibilityRecoveryValue = 0.003;
    private final int recoveryTime = 5;
    private final int vaccineEffectPeriod = 14;

    public double getSusceptibility() {
        return susceptibility;
    }

    public void setSusceptibility(double susceptibility) {
        this.susceptibility = susceptibility;
    }

    public double getTargetSusceptibility() {
        return targetSusceptibility;
    }

    public void setTargetSusceptibility(double targetSusceptibility) {
        this.targetSusceptibility = targetSusceptibility;
    }

    public double getCurrentSusceptibility() {
        return currentSusceptibility;
    }

    public void setCurrentSusceptibility(double currentSusceptibility) {
        this.currentSusceptibility = currentSusceptibility;
    }

    public boolean isInfected() {
        return infected;
    }

    public void setInfected(boolean infected) {
        this.infected = infected;
    }

    public boolean isVaccineEffectCompleted() {
        return vaccineEffectCompleted;
    }

    public void setVaccineEffectCompleted(boolean vaccineEffectCompleted) {
        this.vaccineEffectCompleted = vaccineEffectCompleted;
    }

    public int getInfectionTime() {
        return infectionTime;
    }

    public void setInfectionTime(int infectionTime) {
        this.infectionTime = infectionTime;
    }

    public Set<Integer> getNeighbours() {
        return neighbours;
    }

    public void setNeighbours(Set<Integer> neighbours) {
        this.neighbours = neighbours;
    }

    public Node() {
        random = new Random();
        susceptibility = (double) (random.nextInt(35) / 1000.00) + 0.016;
        currentSusceptibility = targetSusceptibility = susceptibility;
        infected = false;
        vaccineEffectCompleted = true;
        infectionTime = 0;
        neighbours = new HashSet<Integer>();
        System.out.println("Value of susceptibility for node = " + susceptibility);
    }

    public void vaccinate() {
        vaccineEffectCompleted = false;
        targetSusceptibility = susceptibility * (1.0 - vaccineEffectiveness);
    }

    public boolean updateSusceptibility() {
        if(!vaccineEffectCompleted) {
            if(currentSusceptibility > targetSusceptibility) {
                currentSusceptibility -= ((susceptibility * vaccineEffectiveness)/ ((double)vaccineEffectPeriod));
                return true;
            }
            vaccineEffectCompleted = true;
            targetSusceptibility = susceptibility;
        }
        if (currentSusceptibility < targetSusceptibility) {
            currentSusceptibility += susceptibilityRecoveryValue;
            if(currentSusceptibility > targetSusceptibility) {
                currentSusceptibility = targetSusceptibility;
            }
            return true;
        }
        return false;
    }

    public boolean infect(double globalDecayValue) {
        if(isInfected()) return false;
        if (((double) random.nextInt(1001) / 1000.00) < getCurrentSusceptibility()) {
            setInfected(true);
            setInfectionTime(0);
            return true;
        }
        return false;
    }

    public boolean isRecovered() {
        if(!isInfected()) return false;
        infectionTime++;
        if (infectionTime < recoveryTime) {
            return false;
        }
        setInfected(false);
        resetInfection();
        return  true;
    }

    public void resetInfection() {
        infectionTime = 0;
        currentSusceptibility = currentSusceptibility * (1.0 - infectionImmunity);
        targetSusceptibility = susceptibility;
        vaccineEffectCompleted = true;
    }
}